src/internal.c

```diff
@@ -7592,6 +7674,34 @@ static INLINE void CopyNameList(byte* buf, word32* idx,
     *idx = begin;
 }
 
+
+static word32 BuildNameList(char* buf, word32 bufSz,
+        const byte* src, word32 srcSz)
+{
+    const char* name;
+    word32 nameSz, idx;
+
+    idx = 0;
+    do {
+        name = IdToName(*src);
+        nameSz = (word32)WSTRLEN(name);
+
+        if (nameSz + 1 + idx > bufSz) {
+            break;
+        }
+
+        WMEMCPY(buf + idx, name, nameSz);
+        idx += nameSz;
+        buf[idx++] = ',';
+
+        src++;
+        srcSz--;
+    } while (srcSz > 0);
+
+    return idx - 1; /* Take off the size of last ','. */
```

Comment:
If we return idx-1, wouldn't the caller expect the null terminator to come right after the last name? But instead, there'll be a comma there. I think we just need to not add the comma on the final iteration. Also perhaps add the null terminator explicitly to the buffer.

Reply:
check